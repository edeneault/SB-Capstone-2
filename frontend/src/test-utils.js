import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
