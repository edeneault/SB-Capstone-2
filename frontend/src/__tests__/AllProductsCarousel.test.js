import React from "react";
import { Provider } from "react-redux";
import "@testing-library/jest-dom/extend-expect";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import AllProductsCarousel from "../components/AllProductsCarousel.js";

const mockStore = configureMockStore([thunk]);

describe("AllProductsComponent test", () => {
  test("renders AllProductsCarousel component", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <AllProductsCarousel />
      </Provider>,
    );

    expect(getByText("~ All Products ~")).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <AllProductsCarousel />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
