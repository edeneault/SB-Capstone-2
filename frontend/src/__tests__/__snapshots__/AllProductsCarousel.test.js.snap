// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProductsComponent test matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="carousel-title m-0 p-3 text-dark fs-1"
  >
    ~ All Products ~
  </h2>
  <div
    class="react-multi-carousel-list  bg-primary fade-in"
  >
    <ul
      class="react-multi-carousel-track "
      style="transition: none; overflow: unset; transform: translate3d(200px,0,0);"
    />
  </div>
</DocumentFragment>
`;
